### ISSUE TITLE
Themes: The thanks modal stomps the thank you screen on premium theme purchase.

### ISSUE THREAD

**Issue by kwight** [Automattic/wp-calypso#8262]
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/design/<site>
2. Purchase a premium theme.
3. When the purchase completes, briefly see a thank you page, before being kicked back to `/design` and the thanks modal.
#### What I expected

The thanks modal shouldn't appear after a premium purchase, allowing the user to see the Calypso thank you screen for purchasing.
#### What happened instead

Calypso went immediately back to `design` after purchase and showed the theme thanks modal instead.
#### Browser / OS version

Chrome 53 on macOS Sierra
#### Screenshot / Video

![thanks](https://cloud.githubusercontent.com/assets/349751/18895555/80b3cac4-84cf-11e6-8a2d-f60db5bd3d15.gif)

**Comment by ockham** [comment]
This one is known, and probably because of the `defer` in [this action](https://github.com/Automattic/wp-calypso/blob/2acf737cc692502ff0af9eeaeb8b67e7471131a7/client/state/themes/actions.js#L253-L270). Unfortunately, it's even more broken without the `defer` -- I tried to remove it a while ago and failed miserably. We decided to live with it for now since we really want to get our current milestone out the door, and were considering this issue relatively minor while requiring quite a bit of research to fix.
